client/src/includes/bulk-actions.js

```diff
@@ -181,21 +181,38 @@ function onClickSelectAllInListing(e) {
  */
 function onClickActionButton(e) {
   e.preventDefault();
-  const url = e.target.getAttribute('href');
-  const urlParams = new URLSearchParams(window.location.search);
+  const currentURL = new URL(window.location.href);
+  const actionURL = new URL(e.target.getAttribute('href'), currentURL);
+
+  // Construct URL parameters for the action view
+  let urlParams;
   if (checkedState.selectAllInListing) {
+    // Preserve query parameters so the server can reconstruct the results
+    // when performing the action on all items in the current listing
+    urlParams = new URLSearchParams(currentURL.searchParams);
     urlParams.append('id', 'all');
     const parentElement = document.querySelector(BULK_ACTIONS_CHECKBOX_PARENT);
     if (parentElement) {
       const parentPageId = parentElement.dataset.bulkActionParentId;
       urlParams.append('childOf', parentPageId);
     }
   } else {
+    // No need to preserve existing query parameters as we're sending specific ids
+    urlParams = new URLSearchParams();
```

Comment:
This isn't really necessary to fix the issue, i.e. we can keep preserving the query parameters to the bulk action view even when we're selecting specific IDs, but I did this to avoid the misdirection as described in the PR.

This means the listing's params won't be preserved for the bulk action view itself, unless you click "select all items in listing" (i.e. `id=all`).

Note that the listing's params are still preserved for the `next` URL regardless.

---

If this doesn't sound like a good idea, we can keep the old behaviour of always passing the listing's params to the bulk action view even if you're only selecting specific items.